transformer/blacklist.py

```diff
@@ -1,25 +1,28 @@
 import os
 import logging
+from typing import Sequence
 
+Blacklist = Sequence[str]
 
-def on_blacklist(url):
-    """
-    Checks for matching URLs in an ignore file (blacklist)
-    from user's current directory.
-    """
+
+def from_file() -> Blacklist:
     blacklist_file = f"{os.getcwd()}/.urlignore"
     try:
         with open(blacklist_file) as file:
-            blacklist = [line.rstrip("\n") for line in file if len(line) > 1]
-
-        for blacklist_item in blacklist:
-            if blacklist_item in url:
-                return True
-
-        return False
-
+            return [line.rstrip("\n") for line in file if len(line) > 1]
     except OSError as err:
         logging.debug(
             "Could not read blacklist file %s. Reason: %s", blacklist_file, err
         )
-        return False
+
+
+def on_blacklist(blacklist: Blacklist, url: str) -> bool:
+    """
+    Checks for matching URLs in an ignore file (blacklist)
+    from user's current directory.
+    """
+    if blacklist:
+        for blacklist_item in blacklist:
+            if blacklist_item in url:
+                return True
+    return False
```

Comment:
Since `blacklist` is a `Sequence`, the only case where `if blacklist` is not visited is when `blacklist` is empty. However, that's redundant with the `for blacklist_item in blacklist` which will also not be visited in that case. I think this code could therefore be simplified by removing the `if blacklist`.